default.nix

```diff
@@ -0,0 +1,8 @@
+with import <nixpkgs> {};
+stdenv.mkDerivation {
+  name = "mttn";
+  nativeBuildInputs = [
+    cargo gdb nasm
```

Comment:
NB: You only need `nasm` for the testsuite, in case there's a different way to specify that. And `gdb` is only a dependency for convenience within the Docker image; it isn't used anywhere in the code.

Reply:
yeah good to note. looking into splitting out `dev` `build` and `test` deps, but for now just keeping in a flat list.